source/source.go

```diff
@@ -100,13 +105,15 @@ func (s *Source) Read(ctx context.Context) (sdk.Record, error) {
 func (s *Source) Teardown(_ context.Context) error {
 	if s.iterator != nil {
 		s.iterator.Stop()
+		// haris: we should explain why do we need to set it to nil
 		s.iterator = nil
 	}
 	return nil
 }
 
 // Ack is called by the conduit server after the record has been successfully processed by all destination connectors
 func (s *Source) Ack(ctx context.Context, tp sdk.Position) error {
+	// haris: so it seems like we are only logging this, we aren't actually ack-ing anything?
```

Comment:
We do not need to send any ack to Google sheets as we poll the Sheets API for data, so there is no data to be ack'd. Mentioned in comment as well. 
Done.

Reply:
Gotcha. I was trying to get at the following point: if it's only logging, and we have no real ack, then we may not even need the logging. But it's trace level anyway, it won't be shown, and if it's too noisy we can remove it later.